### PR TITLE
Add RuboCop support

### PR DESCRIPTION
**Pull request by bpollack**
This introduces RuboCop to Active Merchant, defaulting to the normal
Ruby community guidelines. To grandfather in our existing code, I've
pre-generated a `.rubocop_todo.yml` file, which will allow
gradual migration.

To run RuboCop, you can run `rake rubocop`, but I've also introduced a
test:local Rake target that runs both the unit tests and RuboCop
checks. I've made this be the default Rake task as well, which means
Travis will (if this PR lands) require RuboCop passing for test
approval.

As a proof-of-concept, I went ahead and fixed WhenThen and
YodaConditional style violations, which had sufficiently few violations
to work for demo purposes.

All PayPal-related code is excluded because it crashes RuboCop.

Unit: 3864 tests, 67919 assertions, 0 failures, 0 errors, 0 pendings, 2 omissions, 0 notifications
100% passed

### CODE REVIEW

**Comment by bpollack** [comment]
Hmm. I'm clearly misunderstanding how to get RuboCop into the Gemfile, but I don't get why what I've got isn't working. (Works locally, fails on Travis.)

File: Gemfile [hunk 1]
```diff
@@ -2,6 +2,7 @@ source 'https://rubygems.org'
 gemspec
 
 gem 'jruby-openssl', :platforms => :jruby
+gem 'rubocop', '~> 0.57.2', require: false
```
Comment (varyonic) [inline Gemfile]: We don't require: false, eg. https://github.com/activeadmin/activeadmin/pull/4810/commits/508287fe0d225a0c7f50bd5d50641bf8a00a7c89

Comment (bpollack) [reply Gemfile]: I dropped the `require: false` and it still dies.

File: Rakefile [hunk 2]
```diff
@@ -23,16 +24,21 @@ task :tag_release do
 end
 
 desc "Run the unit test suite"
-task :default => 'test:units'
+task :default => 'test:local'
 task :test => 'test:units'
 
+RuboCop::RakeTask.new
```
Comment (varyonic) [inline Rakefile]: Looks good to me.  FWIW we went with https://github.com/activeadmin/activeadmin/pull/5174/files

**Comment by bpollack** [comment]
@activemerchant/spreedly-runtime This is ready for an approval/rejection at this point; it's passing CI and clean to merge.

**Pull request merged**